arcgis-ios-sdk-samples/Layers/Map image layer tables/MapImageLayerTablesViewController.swift

```diff
@@ -0,0 +1,158 @@
+        serviceRequestsMapImageLayer.loadTablesAndLayers { [weak self] _ in
+            self?.queryFeatures()
+        }
+        return map
+    }
+    
+    /// Make a renderer for the graphics overlay.
+    func makeRenderer () -> AGSSimpleRenderer {
+        let renderer = AGSSimpleRenderer()
+        // Create a symbol for the feature.
+        let selectedRequestSymbol = AGSSimpleMarkerSymbol(style: .circle, color: .cyan, size: 14)
+        // Set the renderer's symbol.
+        renderer.symbol = selectedRequestSymbol
+        return renderer
+    }
+    
+    /// Query features on the first table in the map image layer.
+    func queryFeatures() {
+        // Create query parameters and set its where clause.
+        let nullCommentsParameters = AGSQueryParameters()
+        nullCommentsParameters.whereClause = "requestid <> '' AND comments <> ''"
+        // Set the first table from the map image layer.
+        commentsTable = serviceRequestsMapImageLayer.tables.first
+        // Query features on the feature table with the query parameters and all feature fields.
+        commentsTable?.queryFeatures(with: nullCommentsParameters, queryFeatureFields: .loadAll) { [weak self] result, error in
+            guard let self = self else { return }
+            if let comments = result?.featureEnumerator().allObjects {
+                // Show the records from the service request comments table in the list view control.
+                self.commentsArray = comments
+                // Enable the button after the map and features have been loaded.
+                self.queryBarButtonItem.isEnabled = true
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+    }
+    
+    /// Query related features for the selected feature.
+    func queryCommentsTable(feature: AGSArcGISFeature) {
+        // Get the relationship that defines related service request features for features in the comments table (this is the first and only relationship).
+        let commentsRelationshipInfo = commentsTable?.layerInfo?.relationshipInfos.first
+        // Create query parameters to get the related service request for features in the comments table.
+        let relatedQueryParameters = AGSRelatedQueryParameters(relationshipInfo: commentsRelationshipInfo!)
+        relatedQueryParameters.returnGeometry = true
+        // Query related features for the selected comment and its related query parameters.
+        commentsTable?.queryRelatedFeatures(for: feature, parameters: relatedQueryParameters) { [weak self] results, error in
+            guard let self = self else { return }
+            // Get the first related feature.
+            if let relatedFeature = results?.first?.featureEnumerator().nextObject() as? AGSArcGISFeature {
+                // Load the feature and get its geometry to show as a graphic on the map.
+                relatedFeature.load { error in
```

Comment:
This still needs to be weakly captured again in `relatedFeature.load(completion:)`.

FYI: https://stackoverflow.com/questions/41991467/where-does-the-weak-self-go/41992442#41992442

Reply:
>  If however, the outer closure has already weakly captured self, then the inner closure will already have a weak reference to self, thus adding [weak self] to the inner closure will have no effect.

The response says that adding `[weak self]` to the inner closure will have no effect. Am I understanding correctly?

Reply:
Haha it is a confusion brought by reusing the name `self`. If you change the left `self` on line 126 to `strongSelf`, you'll see what I mean 😏